### PR TITLE
add experimental support for macs with macports installed

### PR DESCRIPTION
**Pull request by linuxgemini**
currently requires symlinking python{version}(-embed).pc to python(embed).pc or python3(-embed).pc under /opt/local/lib/pkgconfig for python linking

### CODE REVIEW

File: client/Makefile [hunk 1]
```diff
@@ -117,8 +125,13 @@ INCLUDES += $(HARDNESTEDLIBINC)
...
 ifneq ($(SKIPLUASYSTEM),1)
-    LUAINCLUDES = $(shell $(PKG_CONFIG_ENV) pkg-config --cflags lua5.2 2>/dev/null)
-    LUALDLIBS = $(shell $(PKG_CONFIG_ENV) pkg-config --libs lua5.2 2>/dev/null)
+    ifdef MACPORTS_PREFIX
+        LUAINCLUDES = $(shell $(PKG_CONFIG_ENV) pkg-config --cflags lua-5.2 2>/dev/null)
+        LUALDLIBS = $(shell $(PKG_CONFIG_ENV) pkg-config --libs lua-5.2 2>/dev/null)
+    else
+        LUAINCLUDES = $(shell $(PKG_CONFIG_ENV) pkg-config --cflags lua5.2 2>/dev/null)
+        LUALDLIBS = $(shell $(PKG_CONFIG_ENV) pkg-config --libs lua5.2 2>/dev/null)
+    endif
```
Comment (iceman1001) [inline client/Makefile:134]: What is the difference that we need to have this IFDEF ?   It looks the same on both sides.

File: doc/md/Installation_Instructions/Mac-OS-X-MacPorts-Installation-Instructions.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    ```bash
+    export C_INCLUDE_PATH="/opt/local/include"
+    export CPLUS_INCLUDE_PATH="/opt/local/include"
+    export LIBRARY_PATH="/opt/local/lib"
+    export LDFLAGS="-L/opt/local/lib"
+    export CFLAGS="-I/opt/local/include"
+    export CPPFLAGS="-isystem/opt/local/include -I/opt/local/include"
+    ```
+
+    other than just modifying PATH variable to `/opt/local/bin:/opt/local/sbin:/usr/local/bin:/usr/local/sbin:$PATH`.
```
Comment (iceman1001) [inline doc/md/Installation_Instructions/Mac-OS-X-MacPorts-Installation-Instructions.md:22]: unclear instruction here.  Do user have to modify PATH or not?

File: doc/md/Installation_Instructions/Mac-OS-X-MacPorts-Installation-Instructions.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    ```
+
+
+## Compile and use the project
+
+To use the compiled client, the only difference is that the Proxmark3 port is `/dev/tty.usbmodemiceman1`, so commands become:
+
+```sh
+proxmark3 /dev/ttyACM0  =>  proxmark3 /dev/tty.usbmodemiceman1
+```
```
Comment (iceman1001) [inline doc/md/Installation_Instructions/Mac-OS-X-MacPorts-Installation-Instructions.md:66]: I think we should add an instruction to use ./pm3  shell script.   I see many users thinking they need to call the client direct themself with a serial port.

**Comment by iceman1001** [comment]
Nice one!
Some minor comments only

File: client/Makefile (hunk 1 above)
Comment (doegox) [reply client/Makefile:134]: THere is a diff, I see a lua-5.2 and a lua5.2

Comment (iceman1001) [reply client/Makefile:134]: Sneaky dash.  Now I see it too.   Thanks!

**Comment by iceman1001** [comment]
Thanks!

**Pull request merged**